lib/ansible/plugins/become/su.py

```diff
@@ -165,4 +165,4 @@ def build_become_command(self, cmd, shell):
         user = self.get_option('become_user') or ''
         success_cmd = self._build_success_command(cmd, shell)
 
-        return "%s %s %s -c %s" % (exe, flags, user, shlex.quote(success_cmd))
+        return ' '.join([exe, shlex.quote(flags), shlex.quote(user), '-c', shlex.quote(success_cmd)])
```

Comment:
I think it might be better to just use `shlex.join` here, instead of `.join` and explicit `shlex.quote`.

Since this is all controller, and `shlex.join` was added in py3.8, would probably be more simple and reliable.  We would just need to ensure that as with `sudo.py` we don't do anything like `prompt = '-p %s'` we should just accumulate a list and do `cmd.extend(['-p', self.prompt])` and then let `shlex.join(cmd)` do all the work itself.